doc/source/dev/contributor/meson.rst

```diff
@@ -137,6 +137,42 @@ To run the tests for a particular submodule(let's say ``optimize``), you can use
   python dev.py -n -s optimize
 
 
+Use GCC and Clang build parallely
+=================================
+
+As discussed, Meson is fully out-of-place, so different builds will not interfere
+with each other. For example, let us build using GCC and Clang.
+
+1. Build with GCC::
+
+    export CC=gcc
+    meson setup build-gcc --prefix=$PWD/install-gcc
+    ninja -C build-gcc
+
+Using the above commands, meson will build with gcc compiler in ``build-gcc`` directory.
+To then install SciPy into the prefix (``install-gcc/``)::
+
+    meson install -C build-gcc
+
+It will then install to ``$PWD/install-gcc/lib/python3.x/site-packages/scipy``.
+
+2. Build with Clang::
+
+    export CC=clang
+    meson setup build-clang --prefix=$PWD/install-clang
+    ninja -C build-clang
```

Comment:
Note that you also need to set `CXX`. So I would do all this in one go:
```
CC=clang CXX=clang++ python dev.py --build-only --build-dir=build-clang
```
And then below it note that Meson will remember the compiler selection for the `build-clang` dir and it cannot be changed, so each future invocation of `python dev.py --build-dir=build-clang` will automatically use Clang.

And then to run tests:
```
python dev.py --build-dir build-clang -s linalg
```